sklearn/utils/estimator_checks.py

```diff
@@ -290,6 +298,9 @@ def check_pipeline_consistency(name, Estimator):
                       random_state=0, n_features=2, cluster_std=0.1)
     X -= X.min()
     y = multioutput_estimator_convert_y_2d(name, y)
+    if name == "SpectralEmbedding":
+        # This data makes sure that the graph is fully connected.
+        X, y = S_X, S_y
```

Comment:
thanks for the comments... This is just a minor confirmation that the arpacks eigsh does not raise `RuntimeError` but instead just gives out random values, which is the root cause of the the current issue at hand...

```
>>> N = np.array([[0,]*5]*5, dtype=float)
>>> print eigsh(N, k=1)
(array([ 0.]), array([[-0.32529456],
       [ 0.29618396],
       [ 0.794161  ],
       [-0.04909478],
       [-0.41636105]]))
>>> print eigsh(N, k=1)
(array([ 0.]), array([[-0.34134676],
       [-0.71507399],
       [ 0.00610712],
       [-0.58006083],
       [ 0.18879544]]))
```